graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/node/GqlJcrNodeImpl.java

```diff
@@ -543,4 +543,73 @@ public GqlAcl getAcl() {
+            return null;
+        } catch (RepositoryException e) {
+            throw new RuntimeException(e);
+        }
+        try {
+            collectUsages(node.getReferences(), references, environment);
+        } catch (RepositoryException e) {
+            //EDP implementation will throw UnsupportedRepositoryOperationException on access to getReferences
+            if (!(e instanceof UnsupportedRepositoryOperationException)) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        List<GqlUsage> result = FilterHelper.filterConnection(references, fieldFilter, environment);
+        if (fieldSorter != null) {
+            result.sort(SorterHelper.getFieldComparator(fieldSorter, FieldEvaluator.forConnection(environment)));
+        }
+
+        return PaginationHelper.paginate(result, u -> PaginationHelper.encodeCursor(u.getNode().getUuid()), arguments);
+    }
+
+    private void collectUsages(PropertyIterator references, Collection<GqlUsage> gqlReferences, DataFetchingEnvironment environment) throws RepositoryException {
+        while (references.hasNext()) {
+            JCRPropertyWrapper reference = (JCRPropertyWrapper) references.nextProperty();
+            JCRNodeWrapper referencingNode = reference.getParent();
+            if (PermissionHelper.hasPermission(referencingNode, environment)) {
+                String name = reference.getName();
+                String locale = reference.getLocale();
+                if (referencingNode.isNodeType("jnt:translation")) {
+                    referencingNode = referencingNode.getParent();
+                } else if (referencingNode.isNodeType("jnt:referenceInField")) {
+                    String s = referencingNode.getProperty("j:fieldName").getValue().getString();
+                    referencingNode = referencingNode.getParent();
+                    NodeIterator it = referencingNode.getI18Ns();
+                    while (it.hasNext()) {
+                        Node langNode = it.nextNode();
+                        String lang = langNode.getProperty("jcr:language").getString();
+                        if (s.endsWith("_" + lang)) {
+                            locale = lang;
+                            name = StringUtils.substringBeforeLast(s, "_" + lang);
+                            break;
+                        }
+                    }
+                }
+                GqlJcrNode gqlReferencingNode = SpecializedTypesHandler.getNode(referencingNode);
+                GqlJcrProperty gqlReference = gqlReferencingNode.getProperty(name, locale, false);
+                GqlUsage usage = new GqlUsage(gqlReferencingNode);
+                if (gqlReference != null) {
+                    gqlReferences.stream().filter(gqlUsage -> gqlUsage.equals(usage)).findFirst().orElse(usage).addUsage(gqlReference);
+                    if(!gqlReferences.contains(usage)) {
+                        gqlReferences.add(usage);
+                    }
+                }
```

Comment:
quite confusing in my opinion .. also don't like the equals/hashcode on GqlUsage which just check the if the node is the same. maybe you can replace with something like 
```suggestion
                    gqlReferences.stream().filter(gqlUsage -> gqlUsage.getNode().getUuid().equals(gqlReferencingNode.getUuid())).findFirst().orElseGet(() -> {
                        GqlUsage usage = new GqlUsage(gqlReferencingNode);
                        gqlReferences.add(usage);
                        return usage;
                    }).addUsage(gqlReference);
```